### PR TITLE
Feature/mkdocs

### PR DESCRIPTION
**Pull request by h4c5**
## ✒️ Context

Documentation thanks to mkdocs

## 🧱 Description of Changes

- `gen_ref_files.py` script in combination of [mkdocs-gen-files](https://oprypin.github.io/mkdocs-gen-files/) in order to automatically generate reference doc for all templates
- minor change in `generate_api_project` (kwargs initialization)
- minor changes to all files containing an incorrect escape sequence such as `/!\ ` (incorrect because `\  ` is an unvalid escape sequence and all backslash must be escaped in python strings

### CODE REVIEW

File: docs/gen_ref_pages.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+
+def pip_install_packages(*packages, editable=True):
+    """Pip install a package in the current env
+
+    Args:
+        editable (bool, optional): If True, install packages in a editable way. Defaults to True.
+    """
+    command = [sys.executable, "-m", "pip", "install"]
+    if editable: command.append("-e")
```
Comment (gsolard) [inline docs/gen_ref_pages.py:72]: Not a big fan of this one-liner (but it is a personal preference)

File: docs/gen_ref_pages.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+        mkdocs_gen_files.set_edit_path(full_doc_path, doc_path)
+
+
+def main():
+    """Main function that generate and install templates and create references doc"""
+    # Create a .mkdocs to store generated templates
+    create_dot_mkdocs()
+
+    # generate and install templates will tracking changed templates
```
Comment (gsolard) [inline docs/gen_ref_pages.py:179]: Will?

**Review by gsolard — APPROVED** [review]

**Review by h4c5 — COMMENTED** [review]
👍

File: docs/gen_ref_pages.py (hunk 2 above)
Comment (h4c5) [reply docs/gen_ref_pages.py:179]: I meant "while"

**Pull request merged**